// ngx-datetime-range-picker/esm2015/lib/ngx-datetime-range-picker.module.js
import { NgModule, Optional, SkipSelf } from "@angular/core";
import { CommonModule } from "@angular/common";
import { FormsModule } from "@angular/forms";
import { MaterialModule } from "./material/material.module";
import { NgxDatetimeRangePickerComponent } from "./ngx-datetime-range-picker.component";
import { ObjNgFor } from "./pipes/objNgFor.pipe";
import { NgxDatetimeRangePickerService } from "./ngx-datetime-range-picker.service";
export class NgxDatetimeRangePickerModule {
    constructor(parentModule) {
        if (parentModule) {
            throw new Error(`ERR_NGX_DATETIME_RANGE_PICKER:
        NgxDatetimeRangePickerModule is already loaded. Import it in the AppModule only`);
        }
    }
    static forRoot() {
        return {
            ngModule: NgxDatetimeRangePickerModule,
            providers: [NgxDatetimeRangePickerService]
        };
    }
}
NgxDatetimeRangePickerModule.decorators = [
    { type: NgModule, args: [{
                declarations: [ObjNgFor, NgxDatetimeRangePickerComponent],
                imports: [CommonModule, FormsModule, MaterialModule],
                exports: [NgxDatetimeRangePickerComponent, MaterialModule]
            },] }
];
NgxDatetimeRangePickerModule.ctorParameters = () => [
    { type: NgxDatetimeRangePickerModule, decorators: [{ type: Optional }, { type: SkipSelf }] }
];
